### PR TITLE
Use StableHLO acos. Update complex acos accuracy tests.

### LINKED ISSUE #36: JAX acos accuracy and correctness
# Resolution

See https://github.com/pearu/functional_algorithms/issues/36#issuecomment-2552171022

# Problem

```
pytest -sv ../functional_algorithms-3/functional_algorithms/tests/test_accuracy.py -x -k test_unary[acos-jax-cpu-complex64]
```
outputs

```
  +inf    EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE
  max     E=====111111111111111111111111111111111111111111111=================================================1=E
  3e38    E=====111111111111111111111111111111111111111111111===================================================E
  7e31    E========1=1111111111111111111111111111111111111111==============================================1====E
  3e28    E======11111=11111111111111111111111111111111111111============================================11=====E
  2e25    E=====1=111111111111111111111111111111111111111111111111111111111111111111111111111111111111111=1=====E
  7e21    E================1111111111111111111111111111111111===================================================E
  3e18    E=========11===111111111111111111111111111111111111====================================1=1111=========E
  2e15    E=================11=111111111111111111111111111111===================================================E
  7e11    E==========11=====1====1111111111111111111111111111============================1====11====11==========E
  3e8     E==========11============11111111111111111111111111==========================1====1=======11==========E
  1e5     E==========11===============11111111111111111111111==============================1========11==========E
  7e1     E==========11=============111=111111111111111111111=======================1===============11==========E
  3e-2    E==========11=============12=1111111111111111111111=====================11=2111===========11==========E
  1e-5    E==========11=============111=111111111111111111111=======================111=11==========11==========E
  7e-9    E==========11=============1===111111111111111111111=========================1=============11==========E
  3e-12   E==========11=============1===111111111111111111111=========================11============11==========E
  1e-15   E==========11=============1===111111111111111111111==========================1============11==========E
  7e-19   E==========11=============1===111111111111111111111=========================1=1===========11==========E
  3e-22   E==========11=============11==111111111111111111111========================11=1===========11==========E
  1e-25   E==========11=============1===111111111111111111111=========================1=1===========11==========E
  6e-29   E==========11=============1===111111111111111111111===========================1===========11==========E
  3e-32   E==========11=============1===111111111111111111111===========================2===========11==========E
  1e-35   E==========11=============1===111111111111111111111===========================1===========11==========E
  1e-45   E==========11=============1===111111111111111111111=======================================11==========E
  0       E==========11=============1===111111111111111111111=======================================11==========E
  -1e-45  EEEEEEEEEEEEEEEEEEEEEEEEEE1===111111111111111111111==========================EEEEEEEEEEEEEEEEEEEEEEEEEE
  -1e-35  E==========11=============1===111111111111111111111===========================1EEEEEEEEEEEEEEEEEEEEEEEE
  -3e-32  E==========11=============1===111111111111111111111===========================2==EEEEEEEEEEEEEEEEEEEEEE
  -6e-29  E==========11=============1===111111111111111111111===========================1====EEEEEEEEEEEEEEEEEEEE
  -1e-25  E==========11=============1===111111111111111111111=========================1=1======EEEEEEEEEEEEEEEEEE
  -3e-22  E==========11=============11==111111111111111111111========================11=1========EEEEEEEEEEEEEEEE
  -7e-19  E==========11=============1===111111111111111111111=========================1=1==========EEEEEEEEEEEEEE
  -1e-15  E==========11=============1===111111111111111111111==========================1============11EEEEEEEEEEE
  -3e-12  E==========11=============1===111111111111111111111=========================11============11==EEEEEEEEE
  -7e-9   E==========11=============1===111111111111111111111=========================1=============11====EEEEEEE
  -1e-5   E==========11=============111=111111111111111111111=======================111=11==========11======EEEEE
  -3e-2   E==========11=============12=1111111111111111111111=====================11=2111===========11========EEE
  -7e1    E==========11=============111=111111111111111111111=======================1===============11==========E
  -1e5    E==========11===============11111111111111111111111==============================1========11==========E
  -3e8    E==========11============11111111111111111111111111==========================1====1=======11==========E
  -7e11   E==========11=====1====1111111111111111111111111111============================1====11====11==========E
  -2e15   E=================11=111111111111111111111111111111===================================================E
  -3e18   E=========11===111111111111111111111111111111111111====================================1=1111=========E
  -7e21   E================1111111111111111111111111111111111===================================================E
  -2e25   E=====1=111111111111111111111111111111111111111111111111111111111111111111111111111111111111111=1=====E
  -3e28   E======11111=11111111111111111111111111111111111111============================================11=====E
  -7e31   E========1=1111111111111111111111111111111111111111==============================================1====E
  -3e38   E=====111111111111111111111111111111111111111111111===================================================E
  min     E=====111111111111111111111111111111111111111111111=================================================1=E
  -inf    EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE
          -inf  -5e30 -2e21 -8e11 -3e2  -1e-7 -5e-1 -2e-2 -8e-3 8e-36 2e-26 5e-17 1e-7  3e2   8e11  2e21  5e30  +inf
```

| ULP-difference |                              z |            jax:acos(z) |         mpmath:acos(z) |          numpy:acos(z) |
| -------------- | ------------------------------ | ---------------------- | ---------------------- | ---------------------- |
|     4294967296 |                           infj |             (nan-infj) |       (1.5707964-infj) |       (1.5707964-infj) |
|     2238032376 |        (-3.4028235e+38-1e-45j) | (3.1415927-89.415985j) | (3.1415927+89.415985j) | (3.1415927+89.415985j) |
|     2238032376 | (3.4028233e+38-3.0382704e-22j) |        (-0-89.415985j) |             89.415985j |             89.415985j |
|     2236155496 |        (-2.6448922e+35-1e-45j) | (3.1415927-82.256256j) | (3.1415927+82.256256j) | (3.1415927+82.256256j) |
|     2236155496 | (2.6448922e+35-3.0382704e-22j) |        (-0-82.256256j) |             82.256256j |             82.256256j |
|     2235211008 |        (-7.2056355e+33-1e-45j) |  (3.1415927-78.65332j) |  (3.1415927+78.65332j) |  (3.1415927+78.65332j) |
|     2235211008 | (7.2056355e+33-3.0382704e-22j) |         (-0-78.65332j) |              78.65332j |              78.65332j |
|     2234260788 |        (-1.9206197e+32-1e-45j) |  (3.1415927-75.02852j) |  (3.1415927+75.02852j) |  (3.1415927+75.02852j) |
|     2234260788 | (1.9206197e+32-1.3936195e-25j) |         (-0-75.02852j) |              75.02852j |              75.02852j |
|     2233305374 |        (-5.0188613e+30-1e-45j) |   (3.1415927-71.3839j) |   (3.1415927+71.3839j) |   (3.1415927+71.3839j) |
|     2233305374 | (5.0188613e+30-1.3936195e-25j) |          (-0-71.3839j) |               71.3839j |               71.3839j |
|     2232368330 |         (-1.406704e+29-1e-45j) | (3.1415927-67.809364j) | (3.1415927+67.809364j) | (3.1415927+67.809364j) |
|     2232368330 |  (1.406704e+29-1.3936195e-25j) |        (-0-67.809364j) |             67.809364j |             67.809364j |
|     2231427800 |        (-3.8906683e+27-1e-45j) |  (3.1415927-64.22153j) |  (3.1415927+64.22153j) |  (3.1415927+64.22153j) |
|     2231427800 | (3.8906683e+27-1.3936195e-25j) |         (-0-64.22153j) |              64.22153j |              64.22153j |
|     2229595892 |        (-1.0579333e+26-1e-45j) | (3.1415927-60.616676j) | (3.1415927+60.616676j) | (3.1415927+60.616676j) |
|     2229595892 |  (1.0579333e+26-6.365765e-29j) |        (-0-60.616676j) |             60.616676j |             60.616676j |
|     2227694102 |        (-2.8126027e+24-1e-45j) |   (3.1415927-56.9893j) |   (3.1415927+56.9893j) |   (3.1415927+56.9893j) |
|     2227694102 |  (2.8126027e+24-6.365765e-29j) |          (-0-56.9893j) |               56.9893j |               56.9893j |
|     2225786959 |        (-7.4015863e+22-1e-45j) |  (3.1415927-53.35171j) | (3.1415927+53.351715j) |  (3.1415927+53.35171j) |
|            ... |                            ... |                    ... |                    ... |                    ... |

### PR DESCRIPTION
**Pull request by pearu**
As in the title.

Fixes https://github.com/pearu/functional_algorithms/issues/36

### CODE REVIEW

**Comment by pearu** [comment]
^ @hawkinsp @jakevdp please review.

**Comment by pearu** [comment]
Re docs/readthedocs.org:jax failure: locally, building docs is successful.

File: tests/lax_test.py [hunk 1]
```diff
@@ -4485,15 +4485,16 @@ def regions_with_inaccuracies_keep(*to_keep):
     elif name == 'tanh':
       regions_with_inaccuracies_keep('ninf', 'pinf', 'ninfj', 'pinfj')
 
-    elif name == 'arccos':
+    elif name == 'arccos' and jax._src.lib.version <= (0, 4, 33):
```
Comment (hawkinsp) [inline tests/lax_test.py:4488]: You can remove the version guard, the minimum is now 0.4.38.

Comment (pearu) [reply tests/lax_test.py]: Removed, thanks!

**Review by hawkinsp — APPROVED** [review]
